libs/domain/product/src/services/adapter/normalizers/facet-rating/facet-rating.normalizer.spec.ts

```diff
@@ -0,0 +1,132 @@
+      RouterService
+    ) as unknown as MockRouterService;
+  });
+
+  afterEach(() => {
+    destroyInjector();
+    vi.clearAllMocks();
+  });
+
+  it('should return normalized rating facet-navigation', () => {
+    facetRatingNormalizer(mockRatingFacet)
+      .pipe(take(1))
+      .subscribe((result) => {
+        expect(result).toEqual({
+          name: mockRatingFacet.localizedName,
+          parameter: mockRatingFacet.config.parameterName,
+          selectedValues: [],
+          valuesTreeLength: mockRatingFacet.max - mockRatingFacet.min + 1,
+          values: Array.from(new Array(5).keys())
+            .map((i) => {
+              const value = i + 1;
+              return {
+                value: String(value),
+                selected: false,
+                count: 0,
+                disabled: false,
+              };
+            })
+            .reverse(),
+        });
+      });
+  });
+
+  describe('when router has "rating[min]" param', () => {
+    beforeEach(() => {
+      routerService.currentQuery = vi.fn().mockReturnValue(
+        of({
+          'rating[min]': '4',
+        })
+      );
+    });
+
+    it('should return normalized rating facet with selected value', () => {
+      facetRatingNormalizer(mockRatingFacet)
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual({
+            name: mockRatingFacet.localizedName,
+            parameter: mockRatingFacet.config.parameterName,
+            selectedValues: ['4'],
+            valuesTreeLength: mockRatingFacet.max - mockRatingFacet.min + 1,
+            values: Array.from(new Array(5).keys())
+              .map((i) => {
+                const value = i + 1;
+                return {
+                  value: String(value),
+                  selected: value === 4,
+                  count: 0,
+                  disabled: false,
+                };
+              })
+              .reverse(),
+          });
+        });
+    });
+  });
+
+  describe('when min is greater then 1 and max is less then 5', () => {
+    beforeEach(() => {
+      mockRatingFacet.min = 2;
+      mockRatingFacet.max = 4;
+    });
+
+    it('should return normalized rating facet with disabled options', () => {
+      facetRatingNormalizer(mockRatingFacet)
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual({
+            name: mockRatingFacet.localizedName,
+            parameter: mockRatingFacet.config.parameterName,
+            selectedValues: [],
+            valuesTreeLength: mockRatingFacet.max - mockRatingFacet.min + 1,
+            values: Array.from(new Array(5).keys())
+              .map((i) => {
+                const value = i + 1;
+                return {
+                  value: String(value),
+                  selected: false,
+                  count: 0,
+                  disabled:
+                    value < mockRatingFacet.min || value > mockRatingFacet.max,
```

Comment:
We can't separately check just the first and last values - we need to check the whole object that the normalizer returns.
This conditional is used to generate the "right" object which will be compared with the received one. But I will rewrite it on more readable way

Reply:
But we should be sure that other properties are not changed to avoid unexpected side effects.
For example, somebody changed the normalizer logic: when the "max" property < 5 then all options will be disabled. The approach you suggested will pass the test because it tests just the first and last options.

Reply:
I'm just suggesting avoiding conditionals in the tests, the way how to achieve this is up to you. The example above just a short demonstration, you can check each element seperately, ie:

```
expect(result.values[0]).toEqaul(disabledRatingFacet)
expect(result.values[1]).toEqaul(enabledRatingFacet)
...
```

Reply:
Ah, ok, got it. The updated version is already pushed so you can check 🙂